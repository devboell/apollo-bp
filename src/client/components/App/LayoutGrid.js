import styled from 'styled-components'

const LayoutGrid = styled.div`
  display: grid;
  grid-gap: 1em;
  grid-template-columns: 1fr 10fr;
  grid-template-areas:
     "homeIcon header"
     "artistNav artistNav"
     "content content"
`

export default LayoutGrid
