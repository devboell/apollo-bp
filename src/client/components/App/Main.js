import styled from 'styled-components'

const Main = styled.main`
  display: flex;
  justify-content: space-around;
`

export default Main
