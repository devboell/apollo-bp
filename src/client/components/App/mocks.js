
import ARTISTS_QUERY from 'graphql/ArtistsQuery'
import artistsFixt from 'fixtures/artistsQuery'

import ARTIST_BY_PATH_QUERY from 'graphql/ArtistByPathQuery'
import artistByPathFixt from 'fixtures/artistByPathQuery'

export const artists = {
  request: { query: ARTISTS_QUERY },
  result: { data: { artists: artistsFixt } },
}

export const artistByPath = {
  request: { query: ARTIST_BY_PATH_QUERY, variables: { path: 'heroshige' } },
  result: { data: { artistByPath: artistByPathFixt } },
}

export const mocks = [
  artists,
  artistByPath,
]
