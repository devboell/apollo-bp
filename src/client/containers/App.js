import React from 'react'

import artists from 'data/artists.json'
import List from 'components/List'

const App = () =>
  <List {...{ artists }} />

export default App
