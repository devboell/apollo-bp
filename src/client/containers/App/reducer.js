import artists from 'data/artists.json'

const initialState = {
  artists,
}

export default (state = initialState, action) => {
  switch (action.type) {
    default:
      return state
  }
}
