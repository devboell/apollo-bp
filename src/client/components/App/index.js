import React from 'react'

import HomeIconArea from 'components/HomeIconArea'
import HeaderArea from 'components/HeaderArea'
import ArtistNavArea from 'components/ArtistNavArea'
import ContentArea from 'components/ContentArea'

import LayoutGrid from './LayoutGrid'

const App = () => (
  <LayoutGrid>
    <HomeIconArea />

  </LayoutGrid>
)
/*
    <HeaderArea />
    <ArtistNavArea />
    <ContentArea />
*/

export default App
