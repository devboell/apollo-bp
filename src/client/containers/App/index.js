import React from 'react'

import artists from 'data/artists.json'
import Header from 'components/Header'
import ArtistsNav from 'components/ArtistsNav'
import Content from 'components/Content'
import Main from './Main'

const App = () =>
  <div>
    <Header />
    <Main>
      <ArtistsNav {...{ artists }} />
      <Content />
    </Main>
  </div>

export default App
