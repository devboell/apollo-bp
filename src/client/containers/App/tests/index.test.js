import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import initialState from 'data/test/artists.json'
import App from '../index'

const middlewares = []
const mockStore = configureStore(middlewares)
const store = mockStore({ app: { artists: initialState } })

const mountWithRouterEntries = entries =>
  mount(
    <MemoryRouter initialEntries={entries}>
      <Provider store={store}>
        <App />
      </Provider>
    </MemoryRouter>,
    )

describe('App component', () => {
  it('has 4 ListItems, with path /', () => {
    const wrapper = mountWithRouterEntries(['/'])
    expect(wrapper.find('ListItem')).toHaveLength(4)
  })

  it('has 1 Introduction, with path /', () => {
    const wrapper = mountWithRouterEntries(['/'])
    expect(wrapper.find('Introduction')).toHaveLength(1)
  })

  it('has 8 GalleryItems, with path /heroshige', () => {
    const wrapper = mountWithRouterEntries(['/heroshige'])
    expect(wrapper.find('GalleryItem')).toHaveLength(8)
  })

  it('has 1 PaintingViewer, with path /heroshige/sokokura.jpg', () => {
    const wrapper = mountWithRouterEntries(['/heroshige/sokokura.jpg'])
    expect(wrapper.find('PaintingViewer')).toHaveLength(1)
  })
})
