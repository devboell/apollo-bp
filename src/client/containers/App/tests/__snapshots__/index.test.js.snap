// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot 1`] = `
<div>
  <header
    className="gjLlnM"
  >
    <h2>
      Ukiyo-e
    </h2>
  </header>
  <main
    className="jlRHmo"
  >
    <ul
      className="BlXUP"
    >
      <li>
        <a
          href="/Heroshige"
          onClick={[Function]}
        >
          <h3
            className="bfXghp"
          >
            Heroshige
          </h3>
        </a>
      </li>
      <li>
        <a
          href="/Hokusai"
          onClick={[Function]}
        >
          <h3
            className="bfXghp"
          >
            Hokusai
          </h3>
        </a>
      </li>
      <li>
        <a
          href="/Utamaro"
          onClick={[Function]}
        >
          <h3
            className="bfXghp"
          >
            Utamaro
          </h3>
        </a>
      </li>
      <li>
        <a
          href="/Kunisada"
          onClick={[Function]}
        >
          <h3
            className="bfXghp"
          >
            Kunisada
          </h3>
        </a>
      </li>
    </ul>
    <article
      className="jdflTo"
    >
      <div>
        Introduction text
      </div>
    </article>
  </main>
</div>
`;
